### PR TITLE
Improve `bundle add` man page

### PR DESCRIPTION
**Pull request by tnir**
## What was the end-user or developer problem that led to this PR?

It is hard to understand which options can be used at the same time in `bundle add`. Examples look naive.

## What is your fix for the problem, implemented in this PR?

Rewrites bundle-add(1) manpage, for example:

1. `--git` and `--github` are mutually exclusive as CLI options (see the below example)
2. `--branch` and `--ref` are intrinsically exclusive as CLI options (similar to #5855)

The results from `PAGER=cat MANWIDTH=80 exe/bundle help add` would be as follows:

```
BUNDLE-ADD(1)                                                    BUNDLE-ADD(1)

NAME
       bundle-add - Add gem to the Gemfile and run bundle install

SYNOPSIS
       bundle     add     GEM_NAME     [--group=GROUP]     [--version=VERSION]
       [--source=SOURCE]       [--path=PATH]       [--git=GIT|--github=GITHUB]
       [--branch=BRANCH|--ref=REF] [--skip-install] [--strict|--optimistic]

DESCRIPTION
       Adds  the  named gem to the [Gemfile(5)][Gemfile(5)] and run bundle in‐
       stall. bundle install can be avoided by using the flag --skip-install.

OPTIONS
       --version, -v
              Specify version requirements(s) for the added gem.

       --group, -g
              Specify the group(s) for the added gem. Multiple  groups  should
              be separated by commas.

       --source, -s
              Specify the source for the added gem.

       --require, -r
              Adds require path to gem. Provide false, or a path as a string.

       --path Specify the file system path for the added gem.

       --git  Specify the git source for the added gem.

       --github
              Specify the github source for the added gem.

       --branch
              Specify the git branch for the added gem.

       --ref  Specify the git ref for the added gem.

       --skip-install
              Adds the gem to the Gemfile but does not install it.

       --optimistic
              Adds optimistic declaration of version.

       --strict
              Adds strict declaration of version.

EXAMPLES
       1.  You can add rails gem dependency to Gemfile without any version re‐
           striction. The source of the gem would be the global source.

           bundle add rails

       2.  You can add rails gem greater than version 1.1  (not  include  1.1)
           and less than 3.0.

           bundle add rails --version "< 3.0, > 1.1"

       3.  You  can use the custom source https://gems.example.com for the gem
           and also the group for the gem.

           bundle add rails --version "~> 5.0.0" --source  "https://gems.exam‐
           ple.com" --group "development"

       4.  The  following adds the gem entry to Gemfile without installing the
           gem. You can install the gems later via bundle install.

           bundle add rails --skip-install

       5.  You can add the gem into more than one group.

           bundle add rails --group "development, test"

SEE ALSO
       Gemfile(5) https://bundler.io/man/gemfile.5.html, bundle-remove(1) bun‐
       dle-remove.1.html

                                September 2022                   BUNDLE-ADD(1)
```

## 1. `--git` and `--github` are mutually exclusive as CLI options

```
$ bundle add --git https://github.com/rubygems/rubygems --github rubygems/rubygems rubygems

[!] There was an error parsing `injected gems`: The :git option can't be used with `github: "rubygems/rubygems"`. Bundler cannot continue.

 #  from injected gems:1
 #  -------------------------------------------
 >  gem "rubygems", ">= 0", :git => "https://github.com/rubygems/rubygems", :github => "rubygems/rubygems"
 #  -------------------------------------------
```

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [ ] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Comment by jeromedalbert** [comment]
> --branch and --ref are intrinsically exclusive as CLI options (similar to https://github.com/rubygems/rubygems/pull/5855)

--branch and --ref are exclusive for `bundle plugin install`, but they are not currently exclusive for `bundle add`, at least not in practice. Try doing `bundle add devise --github heartcombo/devise --branch main --ref 1d859b5db0dd6cc80322cc9f48706ccf0076ceca` in a Gemfile project, and it currently works with no error.

I think a separate issue could be filed to suggest making --branch and --ref exclusive for `bundle add`, and if this suggestion makes sense, make a PR that would implement a hard fail if both are provided, and that PR would also update `bundle-add`'s manpage accordingly.

In the meantime, this PR is 2 years old and in draft state. Can we close this issue @deivid-rodriguez? If this issue is still of interest to its author, they could open a new issue/PR with the suggestion above.

**Comment by deivid-rodriguez** [comment]
That makes sense, @jeromedalbert.

There's other little improvements that do make sense here, like the part clarifying `github` vs `git`, so let me check with @tnir before closing.

@tnir Are you still interested in moving this forward?

**Comment by deivid-rodriguez** [comment]
The other improvements seemed nice, so I rebased this PR removing the change about `--ref` vs `--branch`.

File: bundler/lib/bundler/man/bundle-add.1.ronn [hunk 1]
```diff
@@ -56,3 +49,33 @@ bundle add rails --group "development, test"
...
+
+   `bundle add rails --version "< 3.0, > 1.1"`
+
+3. You can use the custom source `https://gems.example.com` for the gem and also
+   the group for the gem.
+
+   `bundle add rails --version "~> 5.0.0" --source "https://gems.example.com" --group "development"`
+
+4. The following adds the `gem` entry to Gemfile without installing the gem. You can install the gems
+   later via `bundle install`.
```
Comment (jeromedalbert) [inline bundler/lib/bundler/man/bundle-add.1.ronn:70]:
```suggestion
4. The following adds the `gem` entry to the Gemfile without installing the
   gem. You can install gems later via `bundle install`.
```

File: bundler/lib/bundler/man/bundle-add.1.ronn [hunk 2]
```diff
@@ -56,3 +49,33 @@ bundle add rails --group "development, test"
...
+
+   `bundle add rails --skip-install`
+
+5. You can add the gem into more than one group.
```
Comment (jeromedalbert) [inline bundler/lib/bundler/man/bundle-add.1.ronn:74]:
```suggestion
5. You can assign the gem to more than one group.
```

**Review by jeromedalbert — COMMENTED** [review]
Looks good to me, I have a few minor suggestions to make the sentences and examples flow more.

**Comment by deivid-rodriguez** [comment]
Thanks for the contribution @tnir and for the review @jeromedalbert!

**Comment by tnir** [comment]
Thanks for merging this old proposal after years 😁

**Pull request merged**